### DISCUSSION TITLE
Say hello! 👋

### DISCUSSION THREAD

**Discussion by EwoutH** [mesa/mesa#2465] [General]
# Welcome to Mesa! 👋

Hi everyone! We're excited to meet everyone who is using and contributing to Mesa. Whether you're a long-time user or contributor, just getting started, or simply curious about the project, we'd love to hear from you!

### Join the conversation!
- Tell us a bit about yourself
- Share how you're using Mesa or an exciting project you have done
- Let us know what brought you here
- Feel free to ask questions or share ideas

We like to grow a community around Mesa, get feedback on what we should focus on and make it easier for contributors to get involved. Therefor we would like get to know our users and contributors a bit better, so say hello!

**Comment by EwoutH** [comment]
Let me kick-off, I'm Ewout ter Hoeven, and in three weeks I will graduate from the TU Delft! I started there in 2015, found the right [study](https://www.tudelft.nl/en/education/programmes/bachelors/technische-bestuurskunde/bsc-systems-engineering-policy-analysis-and-management) in 2017, and immediately fell in love with ABM once there. In my bachelors I did two ABM courses (in NetLogo back in the day!), and in 2021 I also started supporting them as teaching assistant. My master was also heavy ABM themed, with two courses, a supporting job and a thesis.

As my Python skills, system knowledge and ABM experience grew, I started slowly contributing small PRs to Mesa. The [first ones](https://github.com/projectmesa/mesa/issues?q=author%3AEwoutH+sort%3Acreated-asc) from 2021 were mainly docs and CI, very small and relatively simple ones. In early 2022 I started feeling comfortable with some larger tuff, while not [everything](https://github.com/projectmesa/mesa/pull/1142) got [merged](https://github.com/projectmesa/mesa/pull/1142) in the end. In the rest of 2022 and early 2023 is was a bit on and off again, depending on how much ABM stuff I did. The first bangers started landing in late 2023, with some of the ones I'm quite proud of:
- https://github.com/projectmesa/mesa/pull/1894
- https://github.com/projectmesa/mesa/pull/1890
- https://github.com/projectmesa/mesa/pull/1898
- https://github.com/projectmesa/mesa/pull/1916

Some where not easy to get in, because of a lot of reasons, but I learned a lot. January 2024 I was asked to join as an maintainer, joining a now 7 member volunteer team that maintains Mesa!

I love how ABMs show sometimes very predictable and sometimes wildly unexpected behavior, from relatively simple components. I also love thinking about how complicated systems actually work, and how we can represent them.

Ourside my modelling, I love spending evenings and weekends away from my computer, with scouting, sailing, surfing, cycling, bouldering, ice skating, the works.

If you have any questions, ask away! (and feel free to drop a way shorter message than mine here)

**Comment by Spartan-71** [comment]
Hello Everyone! 👋

I'm **Anish Dabhane** from India, currently in my 3rd year of a Bachelor's in Computer Science. My passion for **Machine Learning** started in my first year of college, which inspired me to dive into several ML projects. My main focus is **Deep Learning** , which has been a core part of many of my projects involving predictive models and neural networks. 

About six months ago, I began exploring **Reinforcement Learning**, delving into concepts like agents, interaction environments, reward functions, policy gradients, and tools such as OpenAI Gym. During this exploration, I developed a simulation where an AI agent managed a virtual city , gaining hands-on experience with agents and reward functions. I have also had success in competitions. I secured 3rd place in a college **Reinforcement Learning competition** by designing a reward function for *Pocket Tanks* in just 2 days . 

I have also participated in over 7 hackathons (securing top ranks in some) and am deeply passionate about them, as they present exciting challenges and foster teamwork skills.

During **Hacktoberfest**, I discovered some amazing open-source projects and contributed to them, which sparked my interest in open source. This led me to **MESA**, which I came across through the GSoC website. After exploring the documentation, I'm eager to contribute by solving issues and submitting PRs.

Outside of my technical pursuits, I'm a **fitness enthusiast** 💪. 
I enjoy hitting the gym daily, trekking mountains 🏔️, and going for long runs 🏃‍♂️ on weekends.

**Comment by vedpawar2254** [comment]
Hello Mesa Community! :wave:

I’m Ved Pawar, a first-year student majoring in Computer Science and Artificial Intelligence. I’m an open-source enthusiast and contributor, eager to learn and grow.

I’m here seeking guidance and mentorship as I am new here but looking forward to contributing to projects where I can make a meaningful impact while sharpening my skills.
Excited to collaborate, learn, and contribute!

**Reply by Nrd-141** [reply]
Hi I am Nour-ddine, you can contact me. you will find my e-mail in my profile.

**Comment by Rishi2674** [comment]
Hello everyone!
I'm **Rishi Upadhyay**, a pre-final year student studying IT from IIIT Allahabad, India. I have a deep passion for **machine learning** and **Deep Learning** and have been actively doing research in the same. I have also devoted my fair share of time for competitive coding and website development. I have developed several websites and worked on several ML and DL projects in the past.

I started studying **Reinforcement Learning** in July 2024 and have been doing in research in **Swarm Robotics** and have leveraged many **Deep Reinforcement Learning** algorithms and made several projects regarding the same. I have created many multi-agent training environments in Unity and used those environments to train agents on value based algorithms like **DQN, DDQN** and policy based algorithms like **REINFORCE** . I am trying to do some cutting edge research on swarm robotics and am very passionate about make some really good research projects based in this domain.

I came across **Project Mesa** while working on one of my projects and I am also hoping that I could become a contributor for the projects. I am also seeking to learn a lot by contributing in various issues. 
I'm excited to learn, contribute and collaborate!

**Comment by Damrukesh** [comment]
Hey everyone !
Im Damrukesh Daliparthi , 3rd year CSE student from BMSITM Bangalore , India . I have been developing a deep passion and aspiration in AI and ML since the last one year and worked on some basic regression and classification models as of now . I haven't  explored much in open source at all nor did I participate in any hackathon , So I really want to grab this oppurtunity to start my first open source development internship to learn and explore . I have a decent skill in python coding and working with some fun libraries like Tkinter , selenium and some basic API's .

I am also currently laying the groundwork for building multi agent systems with crew ai to build my own projects for fun. When I came across the Mesa organization , I was fascinated and inspired to see many contributors build agents tailoring to different objectives and I realised this is what I really want to explore.

I’m driven to grow my skill set and contribute, and I see mentorship as a key to achieving that.

**Comment by suryanshgargbpgc** [comment]
Hello everyone,
I’m Suryansh Garg, a 3rd-year engineering student at BITS Pilani. I’m currently exploring Python and Django while actively engaging with the open-source community. I’m excited about the opportunity to contribute to MESA and collaborate with like-minded individuals. I look forward to learn from the community, receiving guidance, and growing alongside fellow developers as we work together to make meaningful contributions.      

I request everyone to guide me through this journey.

also @EwoutH is there any tentative idea list using which I can get an understanding for what to focus on 
Thank you

**Comment by umairgcu** [comment]
Dear Mesa Team,

I'm a Python developer with experience in data analysis and a strong interest in agent-based modeling. I'm excited by Mesa's potential for GSoC 2025. I'm particularly interested in projects related to data analysis, visualization, and performance optimization within Mesa.

I'm eager to learn about Mesa's GSoC plans and explore potential contributions. Please let me know if Mesa is considering participation in GSoC 2025.

Thank you for your time and consideration.

**Comment by ashwin-r11** [comment]
# About me

Hi everyone! 👋

I'm **Ashwin R**, a sophomore Computer Science Undergrad from SASTRA University India. I'm passionate in AI, IoT, networking, cybersecurity, and programming. My journey so far has been a blend of science, creativity, and experimentation, and I’m here to explore the exciting world of agent-based simulations using **Mesa**. 🌌

### A bit about me:
- 📚 I’m preparing for **GSoC 2025**, with a focus on projects blending AI, math, and scientific computing.
- 🖥️ My programming toolkit includes C, C++, Java, Python, along with tools like MATLAB, Hugo for web hosting.
- 🌟 I enjoy exploring sci-fi, physics, math, and thought-provoking literature, which often inspires my projects and ideas.
- 🧮 I absolutely love **math and physics**, particularly areas involving **probability, statistics**, and mathematical concepts tied to **predictions and estimations in AI**. These interests fuel my curiosity for modelling real-world systems and creating intelligent solutions. 

I’m here to share my learning journey with **Mesa** and would love to contribute to the community with sample projects and discussions. Feel free to connect if you share similar interests or if you’re also working on something cool with Mesa!

Looking forward to learning, collaborating, and building! 🚀

Cheers,  
**Ashwin R**

**Comment by aliboukind** [comment]
Hi! What is the latest stable version of Mesa. Would it be 3.0.3 or 3.13? Thanks!

**Reply by PrashantChoudhary13579** [reply]
v3.1.3

**Comment by WingchunSiu** [comment]
**About Me & Interest in Mesa-LLM Project (GSoC 2025)**  

Hi Mesa Community! 👋  

My name is Michael Siu, a CS & Applied Math sophomore student at USC who loves AI/ML, LLMs, and building cool projects with smart agents. I’m really interested in the **Mesa-LLM project**! I think integrating LLM-powered agents into simulations for decision-making could open up some awesome new possibilities for research and real-world applications, and I'd love to contribute.  

I have experience working with **LLMs and AI-driven decision-making**—experimenting with **prompt engineering, Chain-of-Thought reasoning, and structured inference techniques**. I’ve also worked with **LLM APIs like OpenAI and Hugging Face** and have experience in **AI software development**.  Currently, I’m joining a **research project focused on integrating LLM-powered agents into an anomaly detection library**, exploring how LLMs can assist in anomaly detection workflows. While I haven’t worked with modular AI agents before, I’m really interested in learning more about their design and integration into agent-based modeling.  

Outside of coding, I’m really into **music 🎸 (I played in a band), basketball 🏀, and exploring creative applications of AI**. 

I’d love to get started by contributing to Mesa and helping with the LLM-agent integration! If there are any **small issues or areas where I can contribute**, let me know—I’d be happy to jump in. Looking forward to collaborating with the Mesa community! 🚀  

Best,  
**Michael**

**Reply by Bhavishya-Chaturvedi** [reply]
Hi @WingchunSiu , Did you find any active forums related this project. I have been struggling to find them😅.
Would really appreciate anything..

**Comment by TomatoChocolate12** [comment]
Hey!

I am Yashas, a Computer Science Undergrad at IIIT Hyderabad, who is also doing research in ML4DrugDiscovery. I came across this organization while browsing for an organization I could contribute to for GSoC'25. I found the Mesa's project ideas very fascinating, especially the **Mesa frames upgrade** project. I fact that this project is relatively in its early stages makes me more excited to contribute to it as it would be a great opportunity for me to help add important core features to the module.

I am experienced with python, C and C++. I have worked quite a bit with application-driven Machine Learning and feel confident with this as well. I have also worked with both pandas and polars frameworks, and am familiar with good software integration practices and using git. Though, admittedly I have little experience with cudF, I am going to try to learn and get familiar with it. I have also tried my hand in contributing to open-source organizations before and have thoroughly enjoyed it. This will be my first year trying out for GSoC and hopefully will enjoy the process.

I would love to show you my skills by contributing to the community, please let me know if there are any areas where I could lend a hand and hopefully learn more about the project!

**Comment by Samarpan1122** [comment]
Hi everyone! 👋

I'm Samarpan Mohanty, a first-year undergrad in Computer Engineering student at UNL and a Research Assistant in Dr. Bavarian's lab, where we use machine learning to advance solid-state battery research.

I have experience working with Python, C#, JavaScript, SQL, and Java for projects, and I also enjoy learning new languages just for fun.

I'm particularly interested in open-source contributions, and I aspire to be a GSoC contributor in 2025. Looking forward to learning more about Mesa and engaging with the community! 🚀

**Comment by Bhavishya-Chaturvedi** [comment]
Hi Mesa Community, 👋👋

Bhavishya Chaturvedi this side, pursuing a Bachelor's in AIML. searching for open-source contributions brought me to Mesa.

I got really interested in the **Mesa-LLM project**. I've worked with LLMs and agents, and the concept of making a user-friendly LLM-based agent really intrigued me.

Previously, worked on research papers and have been interested in reinforcemnt learning and deep learning.
It's my first time with GSOC so hoping for the best. 

I would love to collaborate on this project and would really appreciate the community support.✌️

**Comment by soh-123** [comment]
Hello Mesa community,

I'm Sohier Lotfy, Porsuing masters in computer science in Georgia tech. I've been looking for an open source project to contribute and I am very interested in the suggested projects you have, can't wait to sumbit my porposal and get accepted to work on it.

**Comment by seaseao** [comment]
Hello! I'm Christian from the eastern United States. I've gotten interested in mesa because of a work project where we need to build a platform for running agent based models for epidemiological purposes. I'm hoping to find time to get involved in the community and particularly explore how users could map real world or debated disease model data into the Mesa framework!

**Comment by kundan223** [comment]
Hello everyone,
This is Kundan Yadav from DTU, India. I am currently pursuing Electronics and Communication Engineering.

A bit about me:
I am fairly new to open source and I am planning to apply for GSoC 2026. I have made a few contributions to Django, and I was looking for something with a bit of machine learning involved (or related to it). That’s how I ended up here.

I’m excited to get started, and I also wanted to ask if there are any new contributor guidelines or similar resources available that I should go through.

**Comment by ghost** [comment]
Hi everyone, I’m Hardik 👋
I’m a first-year BCA student, studying at Bennett University, Greater Noida,  with basic Python knowledge and I’m new to open source.
I’d love to start with beginner-friendly issues or documentation tasks and learn by contributing.
Looking forward to learning from the community!

**Comment by Rahul-Araganji** [comment]
Hey everyone 
I’m Rahul. I’m currently a second-year BE student.

I’m completely new to agent-based modeling and open-source contributions, but while going through the GSoC orgs list, Mesa immediately caught my attention. The idea of simulating systems, defining simple agent rules, and then watching complex behavior emerge honestly fascinated me. It even made me think about how many real-world systems (and maybe even the world we live in) could be understood as layers of simple interactions leading to complex outcomes.

I’m here to learn, explore Mesa properly, and gradually get involved with the community. Looking forward to learning from all of you and understanding how others approach modeling with Mesa!

**Comment by emharsha1812** [comment]
Hey everyone. My name is Harshwardhan, and I am an AI engineer. I discovered Mesa while going down a rabbit hole that started with Conway’s Game of Life, and I have been exploring it ever since.

I am glad to see Mesa participating in GSoC 2026, and I am looking forward to contributing to the codebase.

Thanks to everyone who has contributed to building and maintaining this project.

**Comment by IlamaranMagesh** [comment]
Hello all,

I'm Maran, applied ML and data professional, currently completing my MSc Computer Science at the University of Bristol. I stumbled upon Mesa when I was exploring ABM, as my interests and research goes into AI implementation in video games (AI as brains for NPCs, game world updation) and AI model optimisations for inference. I'm also looking to participate in GSOC and I'm glad that the two paths meet 😄 

> "The threads of fate have finally begun to intertwine."
\- Igor, Persona 5

I'm new to ABM but know my stuffs in Finite State Machines so, I guess that helps. Looking forward to learning and contributing (and progressing my research) to the project!

**Reply by EwoutH** [reply]
Thanks for reaching out. Some of our ABMs model Agent behavior as finite states (like the classic SIR model), but not sure if that directly makes it a FSM.

You might be interested in:
- https://github.com/mesa/mesa/discussions/3304
- https://github.com/mesa/mesa/discussions/2538

**Comment by Kunal-Somani** [comment]
Hi everyone! I'm Kunal, an undergraduate pursuing dual degrees in Robotics & AI and Data Science.

I've been working heavily with autonomous AI agents recently (building scheduled research assistants and GitHub triage agents using Python, local LLMs, and RAG). This brought me to Mesa!

I'm specifically here because I am very interested in contributing to the Mesa-LLM Iteration project for GSoC 2026. I'm currently reading through the CONTRIBUTING.md and playing around with the introductory Boltzmann Wealth Model to get a feel for the spatial grids.

Looking forward to getting to know the community and tackling some good first issues soon!

**Comment by Harshini2411** [comment]
Hi Mesa Community! 👋

I'm Harshini Kodali, an MS in Engineering Data Science student at the University of Houston (GPA: 3.81).
I've already started contributing to mesa-examples:
- Opened issue #357 (hex_ant relative import bug)
- Opened issue #358 (mesa.visualization broken in 3 examples)
- Raised PR #348 (fix for conways_game_of_life_fast visualization crash) learned a lot from the maintainer feedback on this one!


Background:
- Research Assistant at UH working with Python, ML, and data pipelines
- Full Stack Engineer at Cognizant (Java, Spring Boot, APIs)
- Experience with LLMs, RAG, LangChain, and agentic AI

I'm particularly interested in the Mesa-LLM project for GSoC 2026, as I have hands-on experience building LLM-powered agentic systems. Looking forward to contributing and learning from the community!

**Comment by MemoonaFatima07** [comment]
Hi everyone,

I'm  Memoona, and I'm presently working toward a bachelor's degree in computer science. I'm very interested in computational modeling, machine learning, and Python-based systems. I recently began investigating agent-based modeling as a method of using simulation to investigate complicated systems.

Mesa's method of creating simulations with modular elements like agents, schedulers, and spatial grids caught my attention when I was looking into Python frameworks for agent-based modeling. For experimentation and study, it is especially fascinating to be able to combine simulation results with Python's data analysis ecosystem.

Additionally, I saw that Mesa is currently switching to Solara as its visualization system. I was drawn to this change since it looks like a fantastic chance for new contributors to learn about the changing architecture and contribute to better examples and documentation.

To gain a deeper understanding of the framework, I am currently looking through the Mesa documentation and using the example models. My objective is to progressively contribute by adding examples, better documentation, and eventually essential functionality.

I'm excited to contribute to the project and learn from the community.

**Comment by apfine** [comment]
Hi Mesa Community! 👋

I'm Adarsh Pandey. 

I've already started contributing to mesa-examples:

Raised PR feat: Add Swarm Minefield Navigator (Dynamic Agent Role management and co-operation) #365 (A deterministic spatial navigation model demonstrating macro-node DFS backtracking, dynamic role handoffs, and strict A* topological verification under partial observability).

https://github.com/mesa/mesa-examples/pull/365

My goal is to replace hard coded deterministic navigation with sentient navigation in the project.

Background & experience :

AI & LLM  projects :   I have deep, hands-on experience building and optimizing AI systems.  I work directly with local foundation models (like deploying Whisper for real-time transcription apps and packaging multimodal deepfake detectors with PyTorch/TensorFlow).

Full-Stack Architecture: I build the heavy-duty infrastructure required to serve these models in production using Next.js, FastAPI, and the MERN stack.

Algorithmis   : I live for complex problem-solving. I actively solve questions in algorithmic programming (C++) and love translating heavy graph-traversal logic into clean code.

I'm particularly interested in the mesa-llm project for GSoC 2026. Given my hands-on experience with LLMs, I want to help push the boundaries of what Mesa agents can do - taking them beyond hardcoded, deterministic spatial heuristics and giving them genuine LLM-driven reasoning and semantic memory.

Looking forward to build and learn with you all!

**Comment by satharyasin-hub** [comment]
Hello everyone! 👋

I'm Yasin from India, interested in open-source development and Python-based simulation systems. Recently I started exploring agent-based modeling and discovered Mesa through its GitHub repository and GSoC discussions.

I recently contributed my first pull request to Mesa, where I removed the deprecated `ContinuousSpaceAgent.pos` property and ensured `position` remains the single source of truth for agent positions. It was a great learning experience working with the codebase and understanding the framework design.

I’m excited to continue contributing, learning more about agent-based modeling, and collaborating with the community.

Looking forward to interacting with everyone here!

**Comment by prakritiie** [comment]
hello mesa community!

I’m Prakriti Gupta, a B.Tech student specializing in AI/ML . I have keen interest in machine learning and enjoy building models and exploring AI systems.

Recently, I’ve been working on a deep learning project in collaboration with a hospital, and I’ve also built several projects involving machine learning and LLMs.

I’m new to agent-based modeling, but I’m excited to learn and contribute to Mesa. I’m also interested in design and improving user experience in tools and documentation.

I’d love to start contributing and learning from the community. Looking forward to collaborating with you all!

**Comment by hyndhavamahesh345** [comment]
Hi Mesa community! 👋
I'm **Hyndhava Mahesh**, an AI & ML undergraduate at **Woxsen University**, Hyderabad, India. I'm applying for **GSoC 2026** and really excited to be here!

**A bit about me:**
I'm currently a Research & Product Development Intern at the Woxsen AI Research Centre where I build RAG pipelines, multi-agent AI systems, and LLM applications using Python, LangChain, CrewAI, and Pinecone. I've built production-grade AI systems like DocuMind Enterprise (a RAG-based document intelligence system with 94% citation accuracy) and SympSense AI (a healthcare ML decision support system).

**What brought me here:**
I'm deeply interested in the **Mesa-LLM Integration project for GSoC 2026** - specifically stabilizing and enhancing LLM-powered agent cognitive architectures. My daily work involves building multi-agent pipelines with LangChain and working with LLM APIs (OpenAI, Gemini), so this project is a natural and exciting fit for me.

**What I plan to do:**
I'm currently going through the Mesa tutorial and building my first agent-based model. I plan to contribute it to mesa-examples before the March 31 deadline. I'd love any guidance from mentors or community members on good first issues or areas to explore!

**GitHub**: https://github.com/hyndhavamahesh345
**Email:** hyndhava.mahesh@gmail.com
**LinkedIn:** https://linkedin.com/in/hyndhavamahesh30894a27a

Looking forward to learning from and contributing to this amazing community! 🚀

**Comment by TheMS1314** [comment]
Hey everyone 👋

I've been going through the mesa-examples repo as part of exploring GSoC 2026 and noticed that the Continuous Space section currently has no examples at all. I thought that was a good opportunity to contribute something meaningful.

I've been working on a Boids/Flocking implementation using Mesa 3.x — basically a modern rewrite since the old boid_flockers in the core repo uses the Mesa 2.x API (unique_id, RandomActivation, JupyterViz) which is now deprecated. The new version uses SolaraViz, shuffle_do, and adds a polarization metric that tracks how ordered the flock is over time, which I think makes it a lot more useful as a teaching example.

Before I open a PR I wanted to check a few things:
- Is this something the maintainers would be open to? I didn't want to just drop a PR without checking first
- Is there a specific style or structure you'd prefer for new examples?
- Are there other continuous space models you'd rather see first?

Happy to adjust the approach based on whatever direction makes most sense for the project. Also open to feedback on the model itself if anyone wants to take a look at what I have so far.

Thanks!

**Reply by IlamaranMagesh** [reply]
There are some guidelines for GSoC candidates before opening a PR, if you have checked [this](https://github.com/mesa/mesa/wiki/Google-Summer-of-Code-2026), you can ignore.

> I've been working on a Boids/Flocking implementation using Mesa 3.x...

Correct me if I understood this wrong. The Boids model has been updated with new methods including shuffle_do in the main branch of the repo. So I think it might be a duplicate effort.

**Reply by TheMS1314** [reply]
Thanks for pointing this out, I just checked the main branch and you're right. It's much more updated than I realized. It's using mesa.experimental.continuous_space and the new Scenario system, which I wasn't aware of.
Given that, I think a standalone Boids example in mesa-examples would actually be redundant now.
Would it make more sense to instead contribute something that doesn't exist yet, like extending the Emperor's Dilemma model with Whistleblower agents and a Belief Gap metric? That's something I've been thinking about independently, and there's no equivalent in the main branch.
Happy to discuss either direction, just want to make sure whatever I work on is actually useful.

**Reply by IlamaranMagesh** [reply]
The first right step to think as a modeller. There isn't any good or bad contribution or an example. Have fun modelling and you'll end up with something incredible ;)

**Comment by SumeetDUTTA** [comment]
Hi @EwoutH and everyone, I am Sumeet, a Computer Applications student with a Political Science minor.
I am preparing a GSoC proposal for Mesa, focused on improving agent communication and interaction patterns (175h), with examples/docs as a backup scope.

Current progress:

  - Built 3 evidence models in my learning space:
    - Market auction (request-reply friction)
    - Needs-driven wolf-sheep (behavior arbitration friction)
    - Opinion diffusion (signals/pub-sub friction)
  - Opened a learning-space PR with standardized READMEs and smoke tests
  - Working on my first upstream Mesa docs PR around Action lifecycle and interrupt semantics

I would appreciate feedback on whether my current 175h scope is appropriately narrow and what maintainers would prioritize first (protocol primitives vs docs/examples).

Thanks, excited to contribute.

**Comment by hyndhavamahesh345** [comment]
Okay 👍

**Comment by alaareda121** [comment]
Hi everyone! 
I'm Alaa, a Computer Science student from Egypt at New Mansoura University. I found Mesa through the GSoC 2026 website and I'm really excited about agent-based modeling.
I'm interested in contributing to the Mesa Examples Revival project. I have experience with Python and I'm eager to learn more about Mesa and the open source community.
Looking forward to learning from everyone here!

**Comment by RodionOm** [comment]
Hi everyone,

I recently started contributing to mesa-examples as part of my GSoC preparation and opened a PR adding a rumor spread example with Solara visualization:
https://github.com/projectmesa/mesa-examples/pull/436

While working through several examples locally, I noticed inconsistencies in dependencies, setup instructions, and compatibility across Mesa versions.
I’m currently interested in the “Reviving Mesa-examples” direction and trying to understand what would be the most valuable contribution:
– adding new, well-structured examples  
– or improving reproducibility / standardization of existing ones  

Would appreciate any guidance on where contributions would have the most impact.

**Comment by mrudula1116** [comment]
Hello all this is mrudula , I have completed BSC and MSC in computer science, I am excited and ready to contribute to mesa @EwoutH

**Comment by ahmadrmq** [comment]
Hi everyone! I’m Ahmad, a Computer Engineering student at (BAU) in Jordan.

I’ve been building autonomous AI systems, including Nexus (an autonomous enterprise engineering agent) and Aletheia (a programmatic hallucination detection system). My work with agentic workflows naturally led me to Mesa, as I’m fascinated by how we can scale complex, multi-level simulations using LLMs.

I’m currently preparing a GSoC 2026 proposal for the Mesa-LLM Iteration project. I’m particularly interested in moving the library toward production stability by implementing more robust tool-calling and memory-retrieval patterns.

Looking forward to contributing and learning from the community!

**Comment by sadrasabouri** [comment]
Hey all,

I'm Sadra Sabouri. I'm a CS PhD candidate at University of Southern California. My research focuses on ways in which LLMs can integrate into human workflows. I'm also a core developer in @openscilab where we design open-source software solutions for scientific computing.

I met @jackiekazil at PyCon-US sprint and want to contribute to Mesa (and Mesa-LLM) project(s). Happy to be in this community. Looking forward to learn from you.

**Reply by jackiekazil** [reply]
Welcome! 👋  One of our biggest needs is sorting through an influx of PRs. 
But we welcome help where we can get it!

**Comment by Amrita-creator** [comment]
Hello everyone!

I'm Amrita Chaturvedi, a third-year Computer Science student. My interest in AI started during my second year when I took an AI course at college. Since then, I've been learning machine learning, neural networks, LangChain, LangGraph, and building AI projects along the way.

I recently became interested in open source and discovered MESA through GSoC resources and community discussions. The project really caught my attention, and I'm excited to learn more, contribute, and connect with the community.

Looking forward to learning from everyone here! 😊

**Comment by piush365** [comment]
Hey everyone!
I'm Piush Gogi, BTech CSE undergrad from India, just completed second year. I work mostly in Python and Java, I have built a custom RoBERTa+BiLSTM sentiment analysis model on Marathi political data, an agentic dev tool called DevFlow Agent, and I maintain Diploma Dost, a live open source platform for diploma students in Maharashtra.
Found Mesa while going deep into the Python ML ecosystem. The ML/RL/AI integration direction in Mesa 4 caught my attention immediately as it completely aligns with where I want to go. Just set up Mesa locally, WolfSheep is running. Starting with tutorials and building models before touching any issues. Aiming for GSoC 2027 and want to be a consistent contributor long before that.

**Comment by amrit712** [comment]
Hey guys,
I am Amrit Jain, pursuing BTech CSE from GGSIPU. I am about to start my second year, and I am great with python and would like to start contributing to something that will be the future. And the MESA project caught my eye I would like to start open-source contribution not just issue solving but feature implementation as well so I was wondering if anyone could tell me where I find the discussion of features to be implemented and stuff so I can start contributing asap.

**Comment by ChuckieZOU** [comment]
Hi everyone! 👋
I’m a PhD candidate at the Hong Kong University of Science and Technology, specializing in environmental economics and public policy.
I’m particularly interested in the application of agent-based modeling (ABM) in the social sciences, especially for policy simulation and the analysis of complex policy processes. My current research is exploring how ABM can be used to model interactions among heterogeneous actors and study the dynamic effects of public policies.

My co-supervisor is a professor in Computer Science, so I am also very interested in the integration of large language models (LLMs) with ABM, including how LLMs might enhance agent heterogeneity, decision-making, and policy simulation while maintaining transparency and reproducibility.

I’m still learning and developing my skills with Mesa, and I’m very glad to join this community. I look forward to learning from everyone here, exchanging ideas, and hopefully contributing to discussions and projects in the future!

**Reply by ChuckieZOU** [reply]
I’m also currently exploring opportunities to visit a U.S. university in 2027 as a visiting PhD student. If you know of professors, research groups, or programs working on computational social science, agent-based modeling, policy simulation, or related areas, I would be very grateful for any recommendations. 😊

**Reply by jackiekazil** [reply]
George Mason - Rob Axtell is there. 
https://science.gmu.edu/academics/departments-units/computational-data-sciences/computational-social-science-phd 

Also - welcome!

**Reply by ChuckieZOU** [reply]
Thank you so much, Jackie! I really appreciate the recommendation and the warm welcome.

I’m currently developing my research further in agent-based modeling, and George Mason’s Computational Social Science program seems extremely relevant to my interests. Prof. Axtell’s work is also very closely aligned with the direction I hope to explore. I’ll definitely take a closer look at his work and the program, and I’m planning to reach out to him as well.

Thanks again for pointing me in this direction!
